File: plugins/device-info/android/DeviceInfoPlugin.kt
package dev.wefter.bridge

import android.content.Context
import org.json.JSONObject

class DeviceInfoPlugin(context: Context, dispatcher: BridgeDispatcher) : WefterPlugin(context, dispatcher) {
    @WefterMethod
    fun getInfo(payload: JSONObject, callback: (Result<Any>) -> Unit) {
        val result =
                JSONObject().apply {
                    put("platform", "android")
                    put("osVersion", android.os.Build.VERSION.RELEASE)
                }
        resolve(callback, result)
    }
}

File: shells/ios-template/WefterBridge/Plugins/DeviceInfoPlugin.swift
import Foundation
import UIKit

final class DeviceInfoPlugin: WefterPlugin {

    // @WefterMethod
    func getInfo(payload: [String: Any], callback: @escaping (Result<Any, Error>) -> Void) throws {
        resolve(callback, data: [
            "platform": "ios",
            "osVersion": UIDevice.current.systemVersion,
        ])
    }
}
